### PR TITLE
fix(data): resolve dataviewer PR #606 follow-ups for issue #624

### LINKED ISSUE #624: dataviewer frontend follow-ups from PR #606: deferred production fixes + component test coverage
## Context

PR #606 (`test/214-dataviewer-custom-hook-tests`) raised hook coverage in `data-management/viewer/frontend` and addressed inline review feedback. During that review, three items were explicitly deferred to keep PR #606 scoped to test-only changes. This issue tracks them.

PR #606: https://github.com/microsoft/physical-ai-toolchain/pull/606

---

## 1. Component test coverage gap

**Scope**: `data-management/viewer/frontend/src/components/**`

PR #606 lifted hook coverage above the new scoped per-file thresholds, but frontend components remain effectively at 0% coverage. The next-phase work is to add component tests (likely React Testing Library + Vitest, mirroring the hook test conventions established in PR #606) and extend `vitest.config.ts` per-file thresholds to cover the most critical components.

**Suggested next steps**

- Inventory components by usage criticality (annotation editor, dataset table, episode viewer, export dialog).
- Add tests for the highest-impact components first.
- Add scoped per-file coverage thresholds to `data-management/viewer/frontend/vitest.config.ts` once a baseline exists.

---

## 2. Deferred production fix: `cancelExport` no-op bug

**File**: `data-management/viewer/frontend/src/hooks/use-export.ts`

`cancelExport` (lines 69–74) unconditionally calls `setError('Export cancelled')` even when no export is in flight:

```ts
const cancelExport = useCallback(() => {
  cancelRef.current?.()
  cancelRef.current = null
  setIsExporting(false)
  setError('Export cancelled')
}, [])
```

When called with no in-flight export, the underlying `cancelRef.current?.()` is correctly a no-op, but the error state is still mutated, which surfaces a misleading "Export cancelled" message in the UI.

The PR #606 test in `data-management/viewer/frontend/src/hooks/__tests__/use-export.test.ts` ("cancelExport is a no-op when no export is in flight") currently encodes the buggy behavior — it asserts `mockCancel` was not called but also asserts `result.current.error === 'Export cancelled'`.

**Fix**

- Guard the `setError('Export cancelled')` call behind an `isExporting` check (or behind `cancelRef.current` being non-null before the early-clear).
- Update the corresponding test assertion to expect `result.current.error === null` after a no-op cancel.

PR #606 inline reply: https://github.com/microsoft/physical-ai-toolchain/pull/606#discussion_r3180265023

---

## 3. Deferred production fix: CSRF helper context dependency

**File**: `data-management/viewer/frontend/src/lib/api-client.ts` (CSRF token caching path) and consumers in `data-management/viewer/frontend/src/hooks/use-labels.ts`

The CSRF helper currently requires manual context wiring during testing. PR #606 worked around this in `use-labels.test.ts` test setup rather than fixing the helper. The result is that tests using mutation hooks must duplicate non-trivial CSRF/context wiring boilerplate.

**Fix**

- Refactor the CSRF helper so it resolves its context independently (e.g., via the existing `_resetCsrfToken` reset path plus a documented seed helper), removing the per-test wiring burden.
- Simplify `use-labels.test.ts` to drop the workaround once the helper change lands.

PR #606 inline reply: https://github.com/microsoft/physical-ai-toolchain/pull/606#discussion_r3180263961

---

## Cross-references

PR #606 review reply discussion URLs:

- https://github.com/microsoft/physical-ai-toolchain/pull/606#discussion_r3180265977 (typed casts)
- https://github.com/microsoft/physical-ai-toolchain/pull/606#discussion_r3180266972 (fetchStatus / setTimeout)
- https://github.com/microsoft/physical-ai-toolchain/pull/606#discussion_r3180267940 (missing tests)
- https://github.com/microsoft/physical-ai-toolchain/pull/606#discussion_r3180263961 (CSRF helper gap — tracked here in §3)
- https://github.com/microsoft/physical-ai-toolchain/pull/606#discussion_r3180265023 (cancelExport bug — tracked here in §2)

### PR DESCRIPTION
**Pull request by WilliamBerryiii**
# Pull Request

## Description

Resolves the four follow-up items from PR #606 review for the dataviewer frontend, tracked under issue #624:

1. **`cancelExport` no-op** — `useExport` no longer fires a spurious `"Export cancelled"` error when the user clicks Cancel after the export has already completed; `cancelRef` is cleared in both `onSuccess` and `onError`, and `cancelExport` early-returns when the ref is null.
2. **CSRF coverage for label mutations** — Added a `mutationFetch` wrapper in `src/lib/api-client.ts` that auto-attaches `X-CSRF-Token` and auth headers for non-GET/HEAD methods (caller headers win on collision; method is normalized to upper case). Migrated `useLabels` PUT/POST/DELETE calls (`setEpisodeLabels`, `addLabelOption`, `removeLabelOption`) to it.
3. **Component test baseline** — Added new test files for `AnnotationWorkspace` (plus a separate diagnostics suite), `DataviewerEpisodeList`, and `ExportDialog`; existing `DataviewerEpisodeViewer`, `CameraSelector`, and `ViewerDisplayControls` suites updated to consume the consolidated test utilities.
4. **Test-utils consolidation** — Collapsed three overlapping test helpers (`render-hook.tsx`, `render-hook-with-query.tsx`, `test-utils.tsx`) into a single `src/test-utils/render.tsx` exporting `createTestQueryClient`, `withQueryClient`, `renderWithQuery`, and `renderHookWithProviders`. `installFetchMock({ csrf: true })` now pre-queues a CSRF token response by default, `jsonResponse` returns a real `Response`, and `setup.ts` resets the cached token between tests via `beforeEach(_resetCsrfToken)`. All 10 hook test files migrated to the new import paths.

Per-file coverage thresholds in `vitest.config.ts` were also expanded to cover `src/components/**` plus the four new component test targets.

Affects `data-management/viewer/frontend/` only — no infrastructure, backend, scripts, or documentation changes.

Closes #624

References (do not close): PR #606

## Type of Change

- [x] 🐛 Bug fix (non-breaking change fixing an issue)
- [ ] ✨ New feature (non-breaking change adding functionality)
- [ ] 💥 Breaking change (fix or feature causing existing functionality to change)
- [ ] 📚 Documentation update
- [ ] 🏗️ Infrastructure change (Terraform/IaC)
- [ ] ♻️ Refactoring (no functional changes)

## Component(s) Affected

- [ ] `infrastructure/terraform/prerequisites/` - Azure subscription setup
- [ ] `infrastructure/terraform/` - Terraform infrastructure
- [ ] `infrastructure/setup/` - OSMO control plane / Helm
- [ ] `workflows/` - Training and evaluation workflows
- [ ] `training/` - Training pipelines and scripts
- [ ] `docs/` - Documentation

Affects `data-management/viewer/frontend/` (not enumerated in template).

## Testing Performed

- [ ] Terraform `plan` reviewed (no unexpected changes)
- [ ] Terraform `apply` tested in dev environment
- [ ] Training scripts tested locally with Isaac Sim
- [ ] OSMO workflow submitted successfully
- [ ] Smoke tests passed (`smoke_test_azure.py`)

Frontend test suite verification (this branch):

- `npm run test` — **1083 passed / 1 skipped / 0 failed**, including:
  - `describe('mutationFetch')` block in `api-client.test.ts` with 5 cases (GET skip, HEAD skip, POST/PUT/DELETE/PATCH attach via `it.each`, lower-case method normalization, caller header override).
  - 3 new CSRF assertion tests in `use-labels.test.ts` covering PUT/POST/DELETE.
- `npm run lint` — **0 errors / 42 warnings** (baseline unchanged).
- Coverage thresholds expanded in `vitest.config.ts`:
  - `src/components/**`: 60% stmts / 55% branches / 40% funcs / 60% lines
  - `ExportDialog.tsx`, `DataviewerEpisodeList.tsx`: 70% stmts / 55% branches / 65% funcs / 70% lines
  - `DataviewerEpisodeViewer.tsx`, `AnnotationWorkspace.tsx`: 80% stmts / 70% branches / 75% funcs / 80% lines

## Documentation Impact

- [x] No documentation changes needed
- [ ] Documentation updated in this PR
- [ ] Documentation issue filed

## Bug Fix Checklist

- [x] Linked to issue being fixed (#624)
- [x] Regression test included
- [ ] Justification for no regression test:

## Checklist

- [x] My code follows the [project conventions](copilot-instructions.md)
- [x] Commit messages follow [conventional commit format](instructions/commit-message.instructions.md)
- [x] I have performed a self-review
- [x] Documentation impact assessed above
- [x] No new linting warnings introduced

🐛 - Generated by Copilot

### CODE REVIEW

**Review by katriendg — APPROVED** [review]
Huge one, but looks complete and thorough work. Thanks!
